### PR TITLE
Move `usePaymentActivityData()` to `PaymentActivity` because filters will need it

### LINKED ISSUE #8733: Move usePaymentActivityData() to PaymentActivity from PaymentActivityData because used by filters
`usePaymentActivityData()` currently is called in `PaymentActivityData`.
We need to move it up to `PaymentActivity` because the filters (eg the preset date range dropdown) will need the return value of `usePaymentActivityData()`, for example the `isLoading` so when data is being fetched, the preset date range dropdown can be hidden or disabled.

### PR DESCRIPTION
**Pull request by shendy-a8c**
Fixes https://github.com/Automattic/woocommerce-payments/issues/8733

#### Changes proposed in this Pull Request

* Move `usePaymentActivityData()` call to `<PaymentActivity>` from `<PaymentActivityData>`.

Because we are going to show filters in the Payment Activity widget, the payment activity data fetch (the call to `usePaymentActivityData()`) needs to happen in the same component of the filter, in `<PaymentActivity>`.
This is needed because while `isLoading` true, I like to either disable or hide the filters and it's hard to do if the `isLoading` is within the child element of `<PaymentActivity>`.

note: I realize that with this change, Payments > Overview page will try to fetch payment activity data even though lifetime TPV is 0 where with base branch, the data fetch will only happen only if lifetime TPV > 0.

* Rename component `<PaymentActivityData>` to `<PaymentActivityDataComponent>` so it does not conflict with the `PaymentActivityData` type.

* Use the date start and end from payment activity data returned by the backend to generate the report links.

#### Testing instructions

* Enable the feature flag `_wcpay_feature_payment_overview_widget`.
* Make sure Payments > Overview > Payment Activity widget shows up correctly.

##### Testing error handling when server does not return proper response.

* Edit `includes/admin/class-wc-rest-payments-reporting-controller.php` locally so it returns an error to frontend.

On this [line](https://github.com/Automattic/woocommerce-payments/blob/56f176df3fff5caa5b5df030f3fd93691ec1c5d2/includes/admin/class-wc-rest-payments-reporting-controller.php#L48):
```diff
-		$wcpay_request->set_date_start( $request->get_param( 'date_start' ) );
+		$wcpay_request->set_date_start( 'abc' );
```

* Make sure Payments > Overview > Payment Activity widget disappears after loading animation.

-------------------

- [x] Run `npm run changelog` to add a changelog file, choose `patch` to leave it empty if the change is not significant. You can add multiple changelog files in one PR by running this command a few times. 
- [x] Covered with tests (or have a good reason not to test in description ☝️)
- [ ] Tested on mobile (or does not apply)

**Post merge**

- [ ] Link to testing instructions from [release testing doc](https://github.com/Automattic/woocommerce-payments/wiki/Release-testing-instructions) following [these instructions](https://github.com/Automattic/woocommerce-payments/wiki/How-to-write-good-manual-testing-scenarios) : _Add link here / 'QA Testing Not Applicable'_
- [ ] Add or update [critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Critical-flows) and [testing instructions for critical flows](https://github.com/Automattic/woocommerce-payments/wiki/Testing-instructions-for-critical-flows), if applicable.
- [ ] Add what's changed (description, screenshot, demo videos etc.) to the release announcement post, if applicable.

### CODE REVIEW

**Comment by botwoo** [comment]
### Test the build 

#### Option 1. Jetpack Beta

- Install and activate [Jetpack Beta](https://jetpack.com/download-jetpack-beta/).
- Use this build by searching for PR number `8751` or branch name `update/8733-move-payment-activity-data-fetching` in your-test.site/wp-admin/admin.php?page=jetpack-beta&plugin=woocommerce-payments          

#### Option 2. Jurassic Ninja - available for logged-in A12s

:rocket: [Launch a JN site with this branch](https://jurassic.ninja/create/?jetpack-beta&shortlived&nojetpack&woocommerce&woocommerce-payments-dev-tools&branches.woocommerce-payments=update/8733-move-payment-activity-data-fetching) :rocket:

:information_source: Install this [Tampermonkey script](https://github.com/Automattic/woocommerce-payments/tree/develop/bin/wcpay-live-branches) to get more options.

---

Build info: 

- Latest commit: ecf0895defc3a68824a600d2ed2debe1a8e492cb
- Build time: 2024-05-21 01:23:27 UTC

_Note: the build is updated when a new commit is pushed to this PR._

File: client/components/payment-activity/index.tsx [hunk 1]
```diff
@@ -5,23 +5,52 @@ import * as React from 'react';
...
 import interpolateComponents from '@automattic/interpolate-components';
+import moment from 'moment';
 
 /**
  * Internal dependencies
  */
 
 import EmptyStateAsset from 'assets/images/payment-activity-empty-state.svg?asset';
-import PaymentActivityData from './payment-activity-data';
+import PaymentActivityDataComponent from './payment-activity-data';
```
Comment (Jinksi) [inline client/components/payment-activity/index.tsx:15]: Kudos for clarifying this component name and fixing the naming conflict with the TS interface.

**Comment by Jinksi** [comment]
> note: I realize that with this change, Payments > Overview page will try to fetch payment activity data even though lifetime TPV is 0 where with base branch, the data fetch will only happen only if lifetime TPV > 0.

@shendy-a8c I think there is an opportunity to separate out the Empty State component to prevent this network call and clarify the logic.

Example:

```tsx
const PaymentActivityWrapper: React.FC = () => {
	const { lifetimeTPV } = wcpaySettings;
	const hasAtLeastOnePayment = lifetimeTPV > 0;

	if ( ! hasAtLeastOnePayment ) {
	// This component can just do empty state.
		return <PaymentActivityEmptyState />;
	}
	
	// This component can call the hook to request API, etc.
	return <PaymentActivity />;
};
```

I've created a PR targeting this branch in #8756 to see if it works or not.

**Update** – merged into this PR.

**Comment by shendy-a8c** [comment]
✓ Resolved conflicts and merged base branch to PR branch.
✓ Handle when server returns an error.
 
Ready for another round of review.

File: client/components/payment-activity/test/__snapshots__/index.test.tsx.snap [hunk 2]
```diff
@@ -76,7 +76,7 @@ exports[`PaymentActivity component should render 1`] = `
               </p>
               <a
                 data-link-type="wc-admin"
-                href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&filter=advanced&date_between%5B0%5D=2024-04-02&date_between%5B1%5D=2024-04-08&search%5B0%5D=charge&search%5B1%5D=payment&search%5B2%5D=payment_failure_refund&search%5B3%5D=payment_refund&search%5B4%5D=refund&search%5B5%5D=refund_failure&search%5B6%5D=dispute&search%5B7%5D=dispute_reversal&search%5B8%5D=card_reader_fee"
+                href="admin.php?page=wc-admin&path=%2Fpayments%2Ftransactions&filter=advanced&date_between%5B0%5D=2024-01-01&date_between%5B1%5D=2024-01-31&search%5B0%5D=charge&search%5B1%5D=payment&search%5B2%5D=payment_failure_refund&search%5B3%5D=payment_refund&search%5B4%5D=refund&search%5B5%5D=refund_failure&search%5B6%5D=dispute&search%5B7%5D=dispute_reversal&search%5B8%5D=card_reader_fee"
```
Comment (Jinksi) [inline client/components/payment-activity/test/__snapshots__/index.test.tsx.snap:79]: It took me a while to realise this change in `date_between` query param values was not a bug – it is simply now using the mock return value.

https://github.com/Automattic/woocommerce-payments/blob/ecf0895defc3a68824a600d2ed2debe1a8e492cb/client/components/payment-activity/test/index.test.tsx#L47-L48

File: client/components/payment-activity/index.tsx [hunk 3]
```diff
@@ -5,57 +5,94 @@ import * as React from 'react';
...
 import './style.scss';
 
+/**
+ * This will be replaces in the future with a dynamic date range picker.
+ */
+const getDateRange = (): DateRange => {
+	return {
+		// Subtract 7 days from the current date.
+		date_start: moment()
+			.subtract( 7, 'd' )
```
Comment (Jinksi) [inline client/components/payment-activity/index.tsx:29]: This date difference was previously set to 6 days. I'm assuming this was intentional, but want to double-check with you.

**Review by Jinksi — APPROVED** [review]
This looks good @shendy-a8c. I've tested the UI in empty state, network error state and happy-path state. All works well.

I've left some minor comments, not merge-blocking.

File: client/components/payment-activity/index.tsx (hunk 3 above)
Comment (shendy-a8c) [reply client/components/payment-activity/index.tsx:29]: Yes that is intentional especially after got confirmation from this discussion p1714639411758769-slack-C06BBFTF6EM where 'last 7 days' should mean excluding today, go back 7 days.

So if today is 2024-05-21, the date start should be 2024-05-14 and date end should be 2024-05-20.

However, now I realize that the date end returned by this `getDateRange()` function is wrong. It should be yesterday (`moment().subtract( 1, 'd' ).format( 'YYYY-MM-DD\\THH:mm:ss' )) and not today (`moment().format( 'YYYY-MM-DD\\THH:mm:ss' )`).

Anyway, this `getDateRange()` function will be removed and I'll make sure to do it right in https://github.com/Automattic/woocommerce-payments/pull/8730.

**Pull request merged**